Auth0Tests/Generators.swift

```diff
@@ -0,0 +1,198 @@
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Foundation
+import JWTDecode
+
+@testable import Auth0
+
+// MARK: - Keys
+
+enum TestKeys {
+    static let rsaPrivate: SecKey = {
+        let query: [String: Any] = [
+            String(kSecAttrKeyType): kSecAttrKeyTypeRSA,
+            String(kSecAttrKeySizeInBits): 2048,
+        ]
+        
+        return SecKeyCreateRandomKey(query as CFDictionary, nil)!
+    }()
+    
+    static let rsaPublic: SecKey = {
+        return SecKeyCopyPublicKey(rsaPrivate)!
+    }()
+}
+
+// MARK: - JWT
+
+private func encodeJWTPart(from dict: [String: Any]) -> String {
+    let json = try! JSONSerialization.data(withJSONObject: dict, options: JSONSerialization.WritingOptions())
+    
+    return json.a0_encodeBase64URLSafe()!
+}
+
+private func generateJWTHeader(alg: String, kid: String) -> String {
+    let headerDict: [String: Any] = ["alg": alg, "kid": kid]
+    
+    return encodeJWTPart(from: headerDict)
+}
+
+private func generateJWTBody(alg: String,
+                             kid: String,
+                             iss: String?,
+                             sub: String?,
+                             aud: [String]?,
+                             exp: Date?,
+                             iat: Date?,
+                             azp: String?,
+                             nonce: String?,
+                             maxAge: Int?,
+                             authTime: Date?) -> String {
+    var bodyDict: [String: Any] = [:]
+    
+    if let iss = iss {
+        bodyDict["iss"] = iss
+    }
+    
+    if let sub = sub {
+        bodyDict["sub"] = sub
+    }
+    
+    if let aud = aud {
+        bodyDict["aud"] = aud.count == 1 ? aud[0] : aud
+    }
+    
+    if let exp = exp {
+        bodyDict["exp"] = exp.timeIntervalSince1970
+    }
+    
+    if let iat = iat {
+        bodyDict["iat"] = iat.timeIntervalSince1970
+    }
+    
+    if let azp = azp {
+        bodyDict["azp"] = azp
+    }
+    
+    if let maxAge = maxAge {
+        bodyDict["max_age"] = maxAge
+    }
+    
+    if let authTime = authTime {
+        bodyDict["auth_time"] = authTime.timeIntervalSince1970
+    }
+    
+    if let nonce = nonce {
+        bodyDict["nonce"] = nonce
+    }
+    
+    return encodeJWTPart(from: bodyDict)
+}
+
+func generateJWT(alg: String = JWTAlgorithm.rs256.rawValue,
+                 kid: String = "key123",
+                 iss: String? = "https://tokens-test.auth0.com/",
+                 sub: String? = "auth0|123456789",
+                 aud: [String]? = ["tokens-test-123"],
+                 exp: Date? = Date().addingTimeInterval(100000),
```

Comment:
magic numbers

Reply:
All the default param values in that method are sane defaults for the tests. What that particular line does is create a date 100000 milliseconds in the future.

Reply:
I get that, but let's give it a name or at least a comment on the side telling what is the intention. e.g. `// tomorrow` or `addingTimeInterval(ONE_DAY_IN_MS)`. Helps improving the readability